src/zope/interface/common/tests/test_collections.py

```diff
@@ -127,6 +128,14 @@ class TestVerifyObject(VerifyObjectMixin,
         'async_generator': unittest.SkipTest,
         type(iter(tuple())): lambda: iter(tuple()),
     }
+    if sys.version_info >= (3, 13):
+        def FrameLocalsProxy_constructor():
+            FrameLocalsProxy = type([sys._getframe().f_locals for x in
+                                     range(1)][0])
+            return FrameLocalsProxy(sys._getframe())
+        FrameLocalsProxy = type([sys._getframe().f_locals for x in
+                                 range(1)][0])
```

Comment:
This code is very difficult to understand.
I suggest the following replacement:
define (early) at module level the function `_get_FrameLocalsProxy` as
```python
def _get_FrameLocalsProxy():
    return type(sys._getframe().f_locals)
```
Replace the code above by
```suggestion
        def FrameLocalsProxy_constructor():
            return _get_FrameLocalsProxy()(sys._getframe())
        FrameLocalsProxy = _get_FrameLocalsProxy()
```

Reply:
Thanks for the simplification idea, I applied that now